quarkus/service/src/main/java/org/apache/polaris/service/quarkus/auth/PolarisAuthenticationMechanism.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.quarkus.auth;
+
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.quarkus.security.credential.TokenCredential;
+import io.quarkus.security.identity.IdentityProviderManager;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.request.AuthenticationRequest;
+import io.quarkus.security.identity.request.TokenAuthenticationRequest;
+import io.quarkus.vertx.http.runtime.security.ChallengeData;
+import io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism;
+import io.quarkus.vertx.http.runtime.security.HttpCredentialTransport;
+import io.quarkus.vertx.http.runtime.security.HttpSecurityUtils;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+import jakarta.annotation.Priority;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Alternative;
+import java.util.Collections;
+import java.util.Set;
+
+/** A custom {@link HttpAuthenticationMechanism} that handles Polaris token authentication. */
+@ApplicationScoped
+@Alternative
+@Priority(1)
+public class PolarisAuthenticationMechanism implements HttpAuthenticationMechanism {
```

Comment:
Should this be under `quarkus/server` perhaps (and the other new auth classes)?

It might be preferable to isolate service code and auth code. Plus, downsteam projects may want to adjust how authentication is down, and adding auth-related beans to the services jar may complicate reuse :thinking:

Reply:
It would cause all tests to fail, since there would be no authentication mechanism present in `polaris-quarkus-service`.

I understand the problem, and indeed I think we should isolate authentication in a dedicated module, but I'd argue that this can be done later. After all, authentication was already being done in `polaris-quarkus-service`, this PR doesn't change this fact.

Reply:
+1 to refactoring in a follow-up PR